src/Umbraco.Core/Services/MemberService.cs

```diff
@@ -254,6 +258,108 @@ public void DeleteMembersOfType(int memberTypeId)
             }
         }
 
+        /// <summary>
+        /// Exports member data based on their unique Id
+        /// </summary>
+        /// <param name="key">The unique <see cref="Guid">member identifier</see></param>
+        /// <param name="currentUser">The <see cref="IUser">user</see> requesting the export</param>
+        /// <returns><see cref="HttpResponseMessage"/></returns>
+        public HttpResponseMessage ExportMemberData(Guid key, IUser currentUser)
+        {
+            var httpResponseMessage = new HttpResponseMessage();
+            if (currentUser.HasAccessToSensitiveData() == false)
+            {
+                httpResponseMessage.StatusCode = HttpStatusCode.Forbidden;
+                return httpResponseMessage;
+            }
+
+            var memberPropertyFilter = new List<string>
+            {
+                "RawPasswordValue",
+                "ParentId",
+                "SortOrder",
+                "Level",
+                "Path",
+                "CreatorId",
+                "Version",
+                "ContentTypeId",
+                "HasIdentity",
+                "PropertyGroups",
+                "PropertyTypes",
+                "ProviderUserKey",
+                "ContentType"
+            };
+            var propertiesFilter = new List<string>
+            {
+                "PropertyType",
+                "Version",
+                "Id",
+                "HasIdentity",
+                "Key"
+            };
+            
+            var member = GetByKey(key);
+            var memberProperties = member.GetType().GetProperties();
+            var fileName = $"{member.Name}_{member.Email}.txt";
+
+            using (var memoryStream = new MemoryStream())
+            {
+                using (var textWriter = new StreamWriter(memoryStream))
+                {
+                    foreach (var memberProperty in memberProperties)
+                    {
+                        if (memberPropertyFilter.Contains(memberProperty.Name))
+                            continue;
+
+                        var propertyValue = memberProperty.GetValue(member, null);

```

Comment:
This needs a null check so a Null ref isn't thrown if the value was null and it's trying to be written to the stream. It can be coalesced like: `var propertyValue = prop.GetValue(owner, null) ?? string.Empty;`